stdlib/benchmarks/builtin/bench_sort_scalars.mojo

```diff
@@ -0,0 +1,238 @@
+@parameter
+fn bench_tiny_list_sort(inout m: Bench) raises:
+    alias counts = List(2, 3, 4, 5)
+
+    @parameter
+    for type_index in range(len(dtypes)):
+        alias dt = dtypes[type_index]
+
+        @parameter
+        for count_index in range(len(counts)):
+            alias count = counts[count_index]
+            var list = random_scalar_list[dt](count)
+
+            @parameter
+            fn bench_sort_list(inout b: Bencher) raises:
+                @always_inline
+                @parameter
+                fn call_fn():
+                    var l1 = List(list)
+                    sort(l1)
+
+                b.iter[call_fn]()
+
+            @parameter
+            fn bench_small_sort(inout b: Bencher) raises:
+                @always_inline
+                @parameter
+                fn call_fn():
+                    var l1 = List(list)
+                    var small_p = rebind[Pointer[Scalar[dt]]](l1.data)
+
+                    @parameter
+                    fn _less_than_equal[
+                        ty: AnyTrivialRegType
+                    ](lhs: ty, rhs: ty) -> Bool:
+                        return rebind[Scalar[dt]](lhs) <= rebind[Scalar[dt]](
+                            rhs
+                        )
+
+                    _small_sort[4, Scalar[dt], _less_than_equal](small_p)
+                    _ = l1
+
+                b.iter[call_fn]()
+
+            @parameter
+            fn bench_insertion_sort(inout b: Bencher) raises:
+                @always_inline
+                @parameter
+                fn call_fn():
+                    var l1 = List(list)
+                    var small_p = rebind[Pointer[Scalar[dt]]](l1.data)
+
+                    @parameter
+                    fn _less_than_equal[
+                        ty: AnyTrivialRegType
+                    ](lhs: ty, rhs: ty) -> Bool:
+                        return rebind[Scalar[dt]](lhs) <= rebind[Scalar[dt]](
+                            rhs
+                        )
+
+                    _insertion_sort[Scalar[dt], _less_than_equal](
+                        small_p, 0, count
+                    )
+                    _ = l1
+
+                b.iter[call_fn]()
+
+            m.bench_function[bench_sort_list](
+                BenchId(
+                    "bench_std_sort_random_list_"
+                    + str(count)
+                    + "_type_"
+                    + str(dt)
+                )
+            )
+            m.bench_function[bench_small_sort](
+                BenchId(
+                    "bench_sml_sort_random_list_"
+                    + str(count)
+                    + "_type_"
+                    + str(dt)
+                )
+            )
+            m.bench_function[bench_insertion_sort](
+                BenchId(
+                    "bench_ins_sort_random_list_"
+                    + str(count)
+                    + "_type_"
+                    + str(dt)
+                )
+            )
+
```

Comment:
```suggestion
            _ = list^
```
We need to manually extend the lifetime of `list`. Otherwise this will not be captured correctly in the parameter closure `call_fn`.